### DISCUSSION TITLE
Has anyone gotten this to work with firebase functions?

### DISCUSSION THREAD

**Discussion by aaronvg** [RobinTail/express-zod-api#210] [Q&A] [answered]
Not sure what I'm doing wrong, tried running this with a local firebase function emulator and couldn't get the routes to attach properly.

**Comment by aaronvg** [comment, marked as answer]
Nevermind got it working.

**Comment by RobinTail** [comment]
Would you like to share your findings in form of some code samples in case someone will be facing the same difficulties in the future? @aaronvg

**Reply by aaronvg** [reply]
Actually when I was testing I had the wrong tsconfig configured (unrelated to this library)
The following code works:
```
const config = createConfig({
  startupLogo: false,
  app: expressApp,
  cors: true,
  logger: {
    level: 'debug',
    color: true,
  },
});

const { notFoundHandler, logger } = attachRouting(config, routing);
expressApp.use(notFoundHandler);

export const app = functions.https.onRequest(expressApp);
```

On a side note, I love this library, but it'd be cool to be able to define the input/outputs separaately (in another package) from the handler functions, so that the backend and frontend code can depend on those types and not have the frontend depend on backend code directly. Anyway not a big deal. Awesome work! You really did some typescript magic with this!
